services/src/main/java/org/keycloak/authentication/authenticators/x509/ValidateX509CertificateUsername.java

```diff
@@ -48,7 +50,7 @@ public void authenticate(AuthenticationFlowContext context) {
         if (certs == null || certs.length == 0) {
             logger.debug("[ValidateX509CertificateUsername:authenticate] x509 client certificate is not available for mutual SSL.");
             context.getEvent().error(Errors.USER_NOT_FOUND);
-            Response challengeResponse = errorResponse(Response.Status.UNAUTHORIZED.getStatusCode(), "invalid_request", "X509 client certificate is missing.");
+            Response challengeResponse = ErrorPage.error(context.getSession(), context.getAuthenticationSession(), Response.Status.UNAUTHORIZED, Messages.X509_CERTIFICATE_MISSING);
```

Comment:
When would this error happen? It seems if there is no certificate x509 authentication should not be attempted?